.github/workflows/test-dfsql.yml

```diff
@@ -0,0 +1,132 @@
+
+      - uses: actions/checkout@v2
+        name: Checkout df-sql
+        with:
+          repository: "oceanprotocol/df-sql"
+          path: "df-sql"
+      
+      - name: Create CSV dir
+        run: |
+          mkdir ~/.dfcsv
+
+      - name: Setup df-sql env file
+        run: |
+          mv df-sql/.env.sample df-sql/.env
+
+      - name: Run df-sql
+        working-directory: ${{ github.workspace }}/df-sql
+        run: |
+          docker-compose up &
+
+      - name: Install dependencies
+        working-directory: ${{ github.workspace }}
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt --quiet
+  
+      - name: Create Brownie paths
+        run: |
+          mkdir -p /home/runner/.solcx
+          mkdir -p /home/runner/.vvm
+          mkdir -p /home/runner/.brownie/packages
+
+      - name: Restore Brownie compilers
+        id: restore-brownie-compilers
+        uses: actions/cache/restore@v3
+        with:
+            path: |
+              /home/runner/.solcx
+              /home/runner/.vvm
+              /home/runner/.brownie/packages
+            key: ${{ runner.os }}-brownie-deps
+
+      - name: Set env variables
+        run: |
+          echo "DFTOOL_KEY=${{ secrets.DFTOOL_KEY_MULTISIG }}" >> $GITHUB_ENV
+          echo "ADDRESS_FILE=.github/workflows/data/address.json" >> $GITHUB_ENV
+          echo "WEB3_INFURA_PROJECT_ID=${{ secrets.WEB3_INFURA_PROJECT_ID }}" >> $GITHUB_ENV
+          echo "SECRET_SEED=${{ secrets.SECRET_SEED }}" >> $GITHUB_ENV
+          echo "date=2023-01-20" >> $GITHUB_ENV
+          echo "now=2023-01-27" >> $GITHUB_ENV
+          echo "CSV_DIR=/tmp/csv" >> $GITHUB_ENV
+          echo "RETRY_TIMES=5" >> $GITHUB_ENV
+
+      - name: Add polygon network
+        run: |
+          brownie networks add polygon polygon host=https://polygon.llamarpc.com chainid=137
+  
+      - name: Run dftool getrate
+        run: |
+          ./dftool getrate OCEAN $date $now $CSV_DIR $RETRY_TIMES
+  
+      - name: Run dftool volsym
+        run: |
+          ./dftool volsym $date $now 2 $CSV_DIR 1 $RETRY_TIMES
+          ./dftool volsym $date $now 2 $CSV_DIR 137 $RETRY_TIMES
+  
+      - name: Run dftool vebals
+        run: |
+          ./dftool vebals $date $now 2 $CSV_DIR 1 $RETRY_TIMES
+  
+      - name: Run dftool allocations
+        run: |
+          ./dftool allocations $date $now 2 $CSV_DIR 1 $RETRY_TIMES
+  
+      - name: Run dftool nftinfo
+        run: |
+          ./dftool nftinfo $CSV_DIR 1 $now
+
+      - name: Run sed
+        run: |
+          cp $CSV_DIR/rate-OCEAN.csv $CSV_DIR/rate-MOCEAN.csv
+          sed -i -e 's/MOCEAN/OCEAN/g' $CSV_DIR/rate-MOCEAN.csv
+  
+      - name: Run dftool calc
+        run: |
+          ./dftool calc $CSV_DIR 100000000 $date OCEAN
+          
+      - name: Move csvs to df-sql dir
+        run: |
+          mv /tmp/csv/* ~/.dfcsv
+        
+      - name: Wait a minute for df-sql to sync
```

Comment:
is there any indication of df-sql "finishing" its work? It seems like an arbitrary number of seconds to wait and maybe we can signal from df-sql that it is somehow done.

Reply:
you can also look into this GH Action to combine these two last steps: https://github.com/marketplace/actions/wait-for-api-action